File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness);
        int n = happiness.length, decay = 0;
        long result = 0L;

        for (int i = n - 1; i >= n - k; i--) {
            result += Math.max(happiness[i] - decay, 0);
            decay += 1;
        }

        return result;
    }
}

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        result = 0
        decay = 0

        for i in range(k):
            result += max(happiness[i] - decay, 0)
            decay += 1

        return result